docs/reasonreactcompat.md

```diff
@@ -0,0 +1,222 @@
+
+```reason
+/* In Image.re */
+[@bs.config {jsx: 3}];
+
+[@react.component]
+let make = (~src) => <img src />;
+
+module Jsx2 = {
+  let component = ReasonReact.statelessComponent("Image");
+  /* `children` is not labelled, as it is a regular parameter in version 2 of JSX */
+  let make = (~src, children) =>
+    ReasonReactCompat.wrapReactForReasonReact(
+      make,
+      makeProps(~src, ()),
+      children,
+    );
+};
+```
+
+Then, `Image` can be used from `Banner`:
+
+```reason
+/* In Banner.re */
+let component = ReasonReact.statelessComponent("Banner");
+
+let make = _children => {
+  ...component,
+  render: _self => <Image.Jsx2 src="./cat.gif" />,
+};
+```
+
+### Component with `children`
+
+Some components pass down `children` transparently, like:
+
+```reason
+/* In SomeButton.re */
+[@react.component]
+let make = (~children) => <button> children </button>;
+```
+
+If this kind of components need to expose a `Jsx2` module for backwards compatibility, like seen above, you might run into errors like:
+
+```
+This expression has type array('a)
+but an expression was expected of type
+ReasonReact.reactElement = React.element
+```
+
+In these cases, it is helpful to wrap `children` with `React.array` inside the `Jsx2` compatibility module, like:
+
+```reason
+/* In SomeButton.re */
+module Jsx2 = {
+  let component = ReasonReact.statelessComponent("SomeButton");
+  let make = children => {
+    let children = React.array(children);
+    ReasonReactCompat.wrapReactForReasonReact(
+      make,
+      makeProps(~children, ()),
+      children,
+    );
+  };
+};
+```
+
+The reason behind those errors is that version 3 of JSX doesn't automatically wrap the `children` passed to an element in an array, like version 2 used to do. Using `React.array` in the `Jsx2` module is a way to provide a consistent behavior for usages of the component in both versions of the JSX transform.
+
+## `wrapReasonReactForReact`: Wrapping a v2 component to be used from a v3 component
+
+Sometimes we might need to make the translation the other way around: wrap a v2 component to be used from a v3 component. For example, if we are using some ReasonReact library that has not been updated yet to be compatible with the latest version.
+
+In these cases, we can use `ReasonReact.wrapReasonReactForReact`.
+
+Let's say we have a v2 component `Text` that needs to be used from a v3 component `Heading`:
+
+```reason
+/* In Text.re */
+let component = ReasonReact.statelessComponent("Text");
+
+let make = (~text, _children) => {
+  ...component,
+  render: _self => <span> {ReasonReact.string(text)} </span>,
+};
+```
+
+We can follow the same approach as above and add a `Jsx3` module to the same file:
+
+```reason
+/* Still in Text.re */
+module Jsx3 = {
+  [@bs.obj] external makeProps: (~text: string, unit) => _ = "";
+  let make =
+    ReasonReactCompat.wrapReasonReactForReact(
+      ~component, (reactProps: {. "text": string}) =>
+      make(~text=reactProps##text, [||])
```

Comment:
can you please also add and example of wrapping component without props?
I've tried a lot of variants, none of them works 😞